hack/build_channel_catalog.sh

```diff
@@ -129,37 +176,28 @@ if [[ ${#skip_versions[@]} -gt 0 ]]; then
     done
 fi
 
-./"$BUNDLE_DEPLOY_DIR"/generate-csv-template.py $generate_csv_template_args > "$csv_template"
-oc process --local -o yaml --raw=true \
-    IMAGE="$OPERATOR_IMAGE" \
-    IMAGE_TAG="$OPERATOR_IMAGE_TAG" \
-    VERSION="$OPERATOR_VERSION" \
-    -f "$csv_template" > "$CSV"
-
-# We need an up-to-date version of opm executable
-opm_local_executable=$(which opm || true)
-if [[ "$opm_local_executable" ]]; then
-    opm_local_version=$(opm version | sed 's/.*OpmVersion:"//;s/".*//')
-fi
-
-if [[ -n "$opm_local_executable" && "$opm_local_version" == "$OPM_VERSION" ]]; then
-    log "Using local opm version $opm_local_executable"
-else
-    opm_download_url="https://github.com/operator-framework/operator-registry/releases/download/$OPM_VERSION/${GOOS}-${GOARCH}-opm"
-    log "Downloading opm from $opm_download_url to $temp_dir/opm"
-    curl -s -L "$opm_download_url" -o "$temp_dir/opm"
-    chmod u+x "$temp_dir/opm"
-    opm_local_executable="$temp_dir/opm"
-fi
+manifests_temp_dir=$(mktemp -d -p "$bundle_temp_dir" manifests.XXXX)
+./hack/generate-operator-bundle-contents.py --name "$OPERATOR_NAME" \
+                                            --version "$OPERATOR_VERSION" \
+                                            --image "$OPERATOR_IMAGE" \
+                                            --image-tag "$OPERATOR_IMAGE_TAG" \
+                                            --output-dir "$manifests_temp_dir" \
+                                            $generate_csv_template_args
 
 # opm won't get anything locally, so we need to push the bundle even in dry run mode
 # we will use a different tag to make sure those leftovers are clearly recognized
 # TODO: remove this tag if we're in dry-run mode in the cleanup trap script
 [[ "$DRY_RUN" == "false" ]] || bundle_image_current_commit="${bundle_image_current_commit}-dryrun"
 log "Creating bundle image $bundle_image_current_commit"
-
-# TODO: use opm to build this
-$CONTAINER_ENGINE build -t "$bundle_image_current_commit" "$BUNDLE_DEPLOY_DIR"
+pushd "$bundle_temp_dir"
+opm alpha bundle build --directory "$manifests_temp_dir" \
+                       --channels alpha \
+                       --default alpha \
```

Comment:
Will this be `alpha` for everyone?

Reply:
Yes. I can move that into a variable, but the fact remains that we will only use one channel and this has to match whatever is in the subscription, e.g. https://github.com/app-sre/deployment-validation-operator/blob/master/deploy/openshift/deployment-validation-operator-olm.yaml#L21

Reply:
fixed in 903ab58